### DISCUSSION TITLE
NUTSHELL: firebaseKey is printing "undefined" in Developer Tools - Sources, but printing correctly in console. (more below)

### DISCUSSION THREAD

**Discussion by hmparsons-sde** [nss-evening-cohort-14/evening-client-side#214] [Help Me!] [answered]
- Have you searched the other help tickets? **Yes**

- If there are similar issues, make sure to give us their number/link to them and tell how yours is different: **N/A**

- Describe your issue: 

I am in the "michelin restaurant" group and am supposed to create a a page that displays all of the reservation data for a single reservation (assigned waitstaff, table, menu items). Starting with assigned staff, I wrote my promises and passed in the additional argument to my printer function. Upon doing that, the new page no longer displays and I get the error below:

<img width="1128" alt="2021-04-11" src="https://user-images.githubusercontent.com/67122062/114316614-e968de00-9ac9-11eb-9e33-714b5b22340c.png">

- What have you googled? (Paste links to the things you have found from your google search - You must include at least 3 links):

[StackOverflow1](-https://stackoverflow.com/questions/60294648/data-from-realtime-firebase-is-displaying-in-console-but-not-in-html-page) (yes, I know this refers to Angular.js)
[StackOverflow2](https://stackoverflow.com/questions/63200955/why-does-one-code-give-me-the-right-key-from-firebase-and-the-other-does-not)
[StackOverflow3](https://stackoverflow.com/questions/51775926/firebase-query-in-react-i-can-log-to-console-but-cannot-use-in-component)
[StackOverflow4](https://stackoverflow.com/questions/45178207/get-firebase-keys-from-snapshot-not-working)

- What have you tried? Tell us everything you have tried.

1. I added an attribute of "firebaseKey" to the link that accesses the single page view. I attempted to target the key this way.
2. I googled the above and tried to find corresponding firebase docs.
3. My husband scoured the code for issues.
4. I had my group members look through it also. 
5. Cried.

- Copy/paste link TO THE FILE on your branch or in your repo that you are having issues with:

[domEvents](https://github.com/nss-evening-cohort-14/michelin-star-restaurant-dashboard-michelin-star-restaurant-dashboard-1/blob/1d5cc7d679a988b5fdc95aea39c7a92b11741036/src/javascripts/events/domEvents.js) **lines 124-143; error at line 128**

[single reservation data](https://github.com/nss-evening-cohort-14/michelin-star-restaurant-dashboard-michelin-star-restaurant-dashboard-1/blob/1d5cc7d679a988b5fdc95aea39c7a92b11741036/src/javascripts/helpers/data/singleReservationData.js) **lines 8-28**

[Single Reservation printer](https://github.com/nss-evening-cohort-14/michelin-star-restaurant-dashboard-michelin-star-restaurant-dashboard-1/blob/1d5cc7d679a988b5fdc95aea39c7a92b11741036/src/javascripts/components/reservations/singleReservation.js) **error thrown at line 6**

[PR Issue](https://github.com/nss-evening-cohort-14/michelin-star-restaurant-dashboard-michelin-star-restaurant-dashboard-1/pull/132)

**Comment by drteresavasquez** [comment, marked as answer]
Working through this...Let us know when you get it resolved.

**Reply by hmparsons-sde** [reply]
10-4, captain!

**Reply by hmparsons-sde** [reply]
Videos #2 and #3 helped a lot. However, the assigned staff is returning undefined. I'm closer, though - the "undefined" is printing on the DOM,  throwing no console errors, and the rest of the information is populating correctly.

**Reply by hmparsons-sde** [reply]
Jesse helped me through this -merging my data required another step where I use Promise.all to map over my staff objects after I map & filter through reservations. This required nested Promises INSIDE of my domEvents block.

**Reply by drteresavasquez** [reply]
@hmparsons-sde share the code block. I wanna see!!

**Reply by hmparsons-sde** [reply]
![Merging Staff Assigned to a reservation](https://user-images.githubusercontent.com/67122062/114785324-2808cd80-9d42-11eb-8085-b1b9a6688174.png)
![Dom Events](https://user-images.githubusercontent.com/67122062/114785314-24754680-9d42-11eb-99c3-4d62f0eccd3c.png)
